### PR TITLE
update to config map to make data value hashed

### PR DESCRIPTION
**Pull request by dbasunag**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new `ConfigMap` class for managing configuration data in Kubernetes.
	- Added functionality for handling binary data and configuration data with validation.
	- Implemented a method for converting `ConfigMap` instances to a dictionary format for API requests.

- **Deprecation Notice**
	- Removed the old `ConfigMap` class and issued a deprecation warning for users to transition to the new module.

### CODE REVIEW

**Comment by redhat-qe-bot2** [comment]
Report bugs in [Issues](https://github.com/myakove/github-webhook-server/issues)

The following are automatically added:
 * Add reviewers from OWNER file (in the root of the repository) under reviewers section.
 * Set PR size label.
 * New issue is created for the PR. (Closed when PR is merged/closed)
 * Run [pre-commit](https://pre-commit.ci/) if `.pre-commit-config.yaml` exists in the repo.

Available user actions:
 * To mark PR as WIP comment `/wip` to the PR, To remove it from the PR comment `/wip cancel` to the PR.
 * To block merging of PR comment `/hold`, To un-block merging of PR comment `/hold cancel`.
 * To mark PR as verified comment `/verified` to the PR, to un-verify comment `/verified cancel` to the PR.
        verified label removed on each new commit push.
 * To cherry pick a merged PR comment `/cherry-pick <target branch to cherry-pick to>` in the PR.
    * Multiple target branches can be cherry-picked, separated by spaces. (`/cherry-pick branch1 branch2`)
    * Cherry-pick will be started when PR is merged
 * To build and push container image command `/build-and-push-container` in the PR (tag will be the PR number).
    * You can add extra args to the Podman build command
        * Example: `/build-and-push-container --build-arg OPENSHIFT_PYTHON_WRAPPER_COMMIT=<commit_hash>`
 * To add a label by comment use `/<label name>`, to remove, use `/<label name> cancel`
 * To assign reviewers based on OWNERS file use `/assign-reviewers`
 * To check if PR can be merged use `/check-can-merge`

<details>
<summary>Supported /retest check runs</summary>

 * `/retest tox`: Retest tox
 * `/retest python-module-install`: Retest python-module-install

</details>

<details>
<summary>Supported labels</summary>

 * hold
 * verified
 * wip
 * lgtm

</details>

**Comment by myakove** [comment]
@dbasunag 
I don't understand how the title of this PR is related to the content of this PR, can you explain?

File: ocp_resources/configmap.py [hunk 1]
```diff
@@ -1,31 +1,9 @@
...
-        super().to_dict()
-
-        if not self.yaml_file and self.data:
-            self.res["data"] = self.data
+from warnings import warn
+from ocp_resources.config_map import ConfigMap  # noqa: F401
+
+warn(
+    f"The module {__name__} is deprecated and will be removed in version 4.17, `ConfigMap` should be imported from "
+    f"`ocp_resources.config_map`",
```
Comment (rnetser) [inline ocp_resources/configmap.py:6]: f-string should be removed

**Review by rnetser — CHANGES_REQUESTED** [review]

**Comment by myakove** [comment]
> @dbasunag I don't understand how the title of this PR is related to the content of this PR, can you explain?

Now I saw this `def keys_to_hash(self):`  and understand

**Review by rnetser — CHANGES_REQUESTED** [review]
Please re-generate the resource - see https://github.com/RedHatQE/openshift-python-wrapper/pull/2121

**Review by rnetser — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Pull request merged**